lib/shim/shim.js

```diff
@@ -949,11 +984,31 @@ function bindSegment(nodule, property, segment, full) {
     property = null
   }
 
-  // This protects against the `bindSegment(func, null, true)` case, where the
-  // segment is `null`, and thus `true` (the full param) is detected as the
-  // segment.
-  if (segment != null && !this.isObject(segment)) {
-    this.logger.debug({ segment: segment }, 'Segment is not a segment, not binding.')
+  const context = this.tracer.getContext()
+  segment = segment || context?.segment
+  const newContext = context.enterSegment({ segment })
+  return this.bindContext({ nodule, property, context: newContext, full })
+}
+
+/**
+ *
+ * @param root0
+ * @param root0.nodule
+ * @param root0.property
+ * @param root0.context
+ * @param root0.full
+ */
+function bindContext({ nodule, property, context, full = false }) {
```

Comment:
The provided doc block is not adding any value. It should be fleshed out, or removed.